### ISSUE TITLE
The config problem of oacnns on s3dis dataset

### ISSUE THREAD

**Issue by woshizhaohaogang** [Pointcept/Pointcept#477]
After reading some of your previous questions and answers on issue, the config on s3dis should refer to the oacnns config on scannet and spunet,According to the voxelsize mentioned in the appendix of your paper and the data preprocessing, authorization and training setting on s3dis，I write the training config corresponding to s3dis as follows

_base_ = ["../_base_/default_runtime.py"]

# misc custom setting
batch_size = 4  # bs: total bs in all gpus
mix_prob = 0.8
empty_cache = False
enable_amp = True
sync_bn = True

# model settings
model = dict(
    type="DefaultSegmentor",
    backbone=dict(
        type="OACNNs",
        in_channels=6,
        num_classes=13,
        embed_channels=64,
        enc_channels=[64, 64, 128, 256],
        groups=[4, 4, 8, 16],
        enc_depth=[3, 3, 9, 8],
        dec_channels=[256, 256, 256, 256],
        point_grid_size=[[8, 12, 16, 16], [6, 9, 12, 12], [4, 6, 8, 8], [3, 4, 6, 6]],
        dec_depth=[2, 2, 2, 2],
        enc_num_ref=[16, 16, 16, 16],
    ),
    criteria=[dict(type="CrossEntropyLoss", loss_weight=1.0, ignore_index=-1)],
)

# scheduler settings
epoch = 3000
optimizer = dict(type="AdamW", lr=0.001, weight_decay=0.05)
scheduler = dict(type="MultiStepLR", milestones=[0.6, 0.8], gamma=0.1)

# dataset settings
dataset_type = "S3DISDataset"
data_root = "/root/autodl-tmp/PT_s3dis"

data = dict(
    num_classes=13,
    ignore_index=-1,
    names=[
        "ceiling",
        "floor",
        "wall",
        "beam",
        "column",
        "window",
        "door",
        "table",
        "chair",
        "sofa",
        "bookcase",
        "board",
        "clutter",
    ],
    train=dict(
        type=dataset_type,
        split=("Area_1", "Area_2", "Area_3", "Area_4", "Area_6"),
        data_root=data_root,
        transform=[
            dict(type="CenterShift", apply_z=True),
            # dict(type="RandomDropout", dropout_ratio=0.2, dropout_application_ratio=0.2),
            # dict(type="RandomRotateTargetAngle", angle=(1/2, 1, 3/2), center=[0, 0, 0], axis="z", p=0.75),
            # dict(type="RandomRotate", angle=[-1, 1], axis="z", center=[0, 0, 0], p=0.5),
            # dict(type="RandomRotate", angle=[-1 / 64, 1 / 64], axis="x", p=0.5),
            # dict(type="RandomRotate", angle=[-1 / 64, 1 / 64], axis="y", p=0.5),
            dict(type="RandomScale", scale=[0.9, 1.1]),
            # dict(type="RandomShift", shift=[0.2, 0.2, 0.2]),
            dict(type="RandomFlip", p=0.5),
            dict(type="RandomJitter", sigma=0.005, clip=0.02),
            # dict(type="ElasticDistortion", distortion_params=[[0.2, 0.4], [0.8, 1.6]]),
            dict(type="ChromaticAutoContrast", p=0.2, blend_factor=None),
            dict(type="ChromaticTranslation", p=0.95, ratio=0.05),
            dict(type="ChromaticJitter", p=0.95, std=0.05),
            # dict(type="HueSaturationTranslation", hue_max=0.2, saturation_max=0.2),
            # dict(type="RandomColorDrop", p=0.2, color_augment=0.0),
            dict(
                type="GridSample",
                grid_size=0.04,
                hash_type="fnv",
                mode="train",
                keys=("coord", "color", "segment"),
                return_grid_coord=True,
            ),
            dict(type="SphereCrop", point_max=100000, mode="random"),
            dict(type="CenterShift", apply_z=False),
            dict(type="NormalizeColor"),
            # dict(type="ShufflePoint"),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                feat_keys=["coord", "color"],
            ),
        ],
        test_mode=False,
    ),
    val=dict(
        type=dataset_type,
        split="Area_5",
        data_root=data_root,
        transform=[
            dict(type="CenterShift", apply_z=True),
            dict(
                type="Copy",
                keys_dict={"coord": "origin_coord", "segment": "origin_segment"},
            ),
            dict(
                type="GridSample",
                grid_size=0.04,
                hash_type="fnv",
                mode="train",
                keys=("coord", "color", "segment"),
                return_grid_coord=True,
            ),
            dict(type="CenterShift", apply_z=False),
            dict(type="NormalizeColor"),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                offset_keys_dict=dict(offset="coord"),
                feat_keys=["coord", "color"],
            ),
        ],
        test_mode=False,
    ),
    test=dict(
        type=dataset_type,
        split="Area_5",
        data_root=data_root,
        transform=[dict(type="CenterShift", apply_z=True), dict(type="NormalizeColor")],
        test_mode=True,
        test_cfg=dict(
            voxelize=dict(
                type="GridSample",
                grid_size=0.04,
                hash_type="fnv",
                mode="test",
                keys=("coord", "color"),
                return_grid_coord=True,
            ),
            crop=None,
            post_transform=[
                dict(type="CenterShift", apply_z=False),
                dict(type="ToTensor"),
                dict(
                    type="Collect",
                    keys=("coord", "grid_coord", "index"),
                    feat_keys=("coord", "color"),
                ),
            ],
            aug_transform=[
                # [dict(type="RandomRotateTargetAngle", angle=[0], axis="z", center=[0, 0, 0], p=1)],
                [dict(type="RandomScale", scale=[0.9, 0.9])],
                [dict(type="RandomScale", scale=[0.95, 0.95])],
                [dict(type="RandomScale", scale=[1, 1])],
                [dict(type="RandomScale", scale=[1.05, 1.05])],
                [dict(type="RandomScale", scale=[1.1, 1.1])],
                [
                    dict(type="RandomScale", scale=[0.9, 0.9]),
                    dict(type="RandomFlip", p=1),
                ],
                [
                    dict(type="RandomScale", scale=[0.95, 0.95]),
                    dict(type="RandomFlip", p=1),
                ],
                [dict(type="RandomScale", scale=[1, 1]), dict(type="RandomFlip", p=1)],
                [
                    dict(type="RandomScale", scale=[1.05, 1.05]),
                    dict(type="RandomFlip", p=1),
                ],
                [
                    dict(type="RandomScale", scale=[1.1, 1.1]),
                    dict(type="RandomFlip", p=1),
                ],
            ],
        ),
    ),
)

But the training effect is not ideal. Is there a problem in the config I wrote myself? And when I training s3dis according to gridsize0.02 in scannet, and open TTA during test, I get stuck during test, so I change it to 0.04,

Another problem is that when I use oacnns to train on scannet, point_grid_size does not use [[8, 12, 16, 16], [6, 9, 12, 12], [4, 6, 8, 8], [3, 4, 6, 6]] in config, but network initialization [[16, 32, 64], [8, 16, 24], [4, 8, 12], [2, 4, 6]], which also reaches 76.1 in the paper.Which do you recommend? And can you tell me if the config given above has errors? I have been working on oacnns recently, and I am looking forward to your reply!

**Comment by pandeng-001** [comment]
Hi, I'm also using OACNNS to validate the paper's results on other datasets. However, the authors only provided the configuration file for ScanNet. I rewrote the configuration file for S3DIS following SPUNet's setup, similar to what you shared.
When running the **test** command, I encountered an out-of-memory (OOM) error, causing the process to be killed by the system. I checked many potential issues but couldn't pinpoint the cause. Eventually, I modified the configuration file by reducing the number of parameters in **config.test_cfg.aug_transform** to around 4, which allowed the **test** to run—though the results weren't great (only ~60% mIoU on Area 5).
Have you faced similar issues? (I'm training and testing on a single 3090 GPU.)

**Comment by woshizhaohaogang** [comment]
> Hi, I'm also using OACNNS to validate the paper's results on other datasets. However, the authors only provided the configuration file for ScanNet. I rewrote the configuration file for S3DIS following SPUNet's setup, similar to what you shared. When running the **test** command, I encountered an out-of-memory (OOM) error, causing the process to be killed by the system. I checked many potential issues but couldn't pinpoint the cause. Eventually, I modified the configuration file by reducing the number of parameters in **config.test_cfg.aug_transform** to around 4, which allowed the **test** to run—though the results weren't great (only ~60% mIoU on Area 5). Have you faced similar issues? (I'm training and testing on a single 3090 GPU.)

With regard to the OOM problem in test, I used a single 4090 training. The problem I ruled out was that gridsize could not be set to 0.02 like scannet's config. I changed it to 0.04 and left the part of aug_transform that was not commented out with #，there will be no OOM problem.
I think you should read the configuration described below and write details of your config again,maybe something subtle went wrong

**Comment by ybc-ybc** [comment]
> Hi, I'm also using OACNNS to validate the paper's results on other datasets. However, the authors only provided the configuration file for ScanNet. I rewrote the configuration file for S3DIS following SPUNet's setup, similar to what you shared. When running the **test** command, I encountered an out-of-memory (OOM) error, causing the process to be killed by the system. I checked many potential issues but couldn't pinpoint the cause. Eventually, I modified the configuration file by reducing the number of parameters in **config.test_cfg.aug_transform** to around 4, which allowed the **test** to run—though the results weren't great (only ~60% mIoU on Area 5). Have you faced similar issues? (I'm training and testing on a single 3090 GPU.)

Hi, could you share how long it took to train oacnns on S3DIS or ScanNet using a 3090 GPU?

**Comment by pandeng-001** [comment]
> > Hi, I'm also using OACNNS to validate the paper's results on other datasets. However, the authors only provided the configuration file for ScanNet. I rewrote the configuration file for S3DIS following SPUNet's setup, similar to what you shared. When running the **test** command, I encountered an out-of-memory (OOM) error, causing the process to be killed by the system. I checked many potential issues but couldn't pinpoint the cause. Eventually, I modified the configuration file by reducing the number of parameters in **config.test_cfg.aug_transform** to around 4, which allowed the **test** to run—though the results weren't great (only ~60% mIoU on Area 5). Have you faced similar issues? (I'm training and testing on a single 3090 GPU.)
> 
> Hi, could you share how long it took to train oacnns on S3DIS or ScanNet using a 3090 GPU?
Hello, sorry for the late reply. I checked the `train.log` file and did some simple calculations. Here's the information for your reference:

| Dataset      | Training Time | GPU       | Epochs |
|--------------|---------------|-----------|--------|
| ScanNet20    | ~31 hours     | Single 3090 | 100    |
| ScanNet200   | ~34 hours     | Single 3090 | 100    |
| S3DIS        | ~15 hours     | Single 3090 | 100    |

**Comment by ybc-ybc** [comment]
> > > Hi, I'm also using OACNNS to validate the paper's results on other datasets. However, the authors only provided the configuration file for ScanNet. I rewrote the configuration file for S3DIS following SPUNet's setup, similar to what you shared. When running the **test** command, I encountered an out-of-memory (OOM) error, causing the process to be killed by the system. I checked many potential issues but couldn't pinpoint the cause. Eventually, I modified the configuration file by reducing the number of parameters in **config.test_cfg.aug_transform** to around 4, which allowed the **test** to run—though the results weren't great (only ~60% mIoU on Area 5). Have you faced similar issues? (I'm training and testing on a single 3090 GPU.)
> > 
> > 
> > Hi, could you share how long it took to train oacnns on S3DIS or ScanNet using a 3090 GPU?
> > Hello, sorry for the late reply. I checked the `train.log` file and did some simple calculations. Here's the information for your reference:
> 
> Dataset	Training Time	GPU	Epochs
> ScanNet20	~31 hours	Single 3090	100
> ScanNet200	~34 hours	Single 3090	100
> S3DIS	~15 hours	Single 3090	100

Thanks a lot！

**Comment by Gofinge** [comment]
Hey all. If GPU memory is not sufficient, adjust grid size to 0.04m is a good choice for S3DIS. We generally use 0.02 as our GPU has large GPU memorys and we want to unify model by multi-datasets joint training with point prompt training.
